src/app/components_ngrx/planner-board/planner-board.component.ts

```diff
@@ -1,10 +1,109 @@
-import { Component } from '@angular/core';
+import {
+  AfterViewChecked,
+  Component,
+  ElementRef,
+  OnDestroy,
+  OnInit,
+  Renderer2,
+  ViewChild
+} from '@angular/core';
+import { ActivatedRoute, NavigationStart, Router } from '@angular/router';
+import { GroupTypeQuery, GroupTypeUI } from './../../models/group-types.model';
+import { IterationQuery } from './../../models/iteration.model';
+import { SpaceQuery } from './../../models/space';
 
 @Component({
     selector: 'planner-board',
     templateUrl: './planner-board.component.html',
     styleUrls: ['./planner-board.component.less']
 })
-export class PlannerBoardComponent {
+export class PlannerBoardComponent implements AfterViewChecked, OnInit, OnDestroy {
+    @ViewChild('boardContainer') boardContainer: ElementRef;
 
+    private uiLockedSidebar: boolean = false;
+    private sidePanelOpen: boolean = true;
+    private eventListeners: any[] = [];
+
+    constructor(
+      private renderer: Renderer2,
+      private spaceQuery: SpaceQuery,
+      private groupTypeQuery: GroupTypeQuery,
+      private iterationQuery: IterationQuery,
+      private route: ActivatedRoute,
+      private router: Router
+    ) {}
+
+    ngOnInit() {
+      this.iterationQuery.deselectAllIteration();
+      this.eventListeners.push(
+        this.spaceQuery.getCurrentSpace
+          .do(() => {
+            // this.uiLockedSidebar = true;
```

Comment:
Don't we need this?

Reply:
Not at this moment.. We can live without  it.

Reply:
Then can you remove this do block?